### PR TITLE
Decouple stdout_syslog and stdout_logfile (see Issue #1375)

### PR DESCRIPTION
**Pull request by scottkof**
This is a fix for [Issue #1375](https://github.com/Supervisor/supervisor/issues/1375)

### CODE REVIEW

File: supervisor/dispatchers.py [hunk 1]
```diff
@@ -139,17 +137,23 @@ def _setup_logging(self, config, channel):
...
                 "Use %s_syslog instead." % channel, DeprecationWarning)
             fmt = ' '.join((config.name, fmt))
-        self.mainlog = loggers.handle_file(
-            config.options.getLogger(),
-            filename=logfile,
-            fmt=fmt,
-            rotating=not not maxbytes, # optimization
-            maxbytes=maxbytes,
-            backups=backups)
+        if logfile is not None:
```
Comment (mnaberez) [inline supervisor/dispatchers.py:140]: This replaces the original comparison `if not logfile` which would evaluate to `False` on both `None`and an empty string.  This data originates from parsing the config file, so it an empty string may actually be possible (unsure).  Was this change intentional, and if so are you certain there is no code path where this may receive an empty string?

File: supervisor/loggers.py [hunk 2]
```diff
@@ -406,6 +406,8 @@ def handle_syslog(logger, fmt):
     handler.setLevel(logger.level)
     logger.addHandler(handler)
 
+    return logger
+
```
Comment (mnaberez) [inline supervisor/loggers.py:410]: It is weird that some of these `handle_` functions return a `logger` and some don't.  Why was this change made?

File: supervisor/tests/test_dispatchers.py [hunk 3]
```diff
@@ -512,6 +512,76 @@ def test_syslog_logfile_deprecated(self):
             self._makeOne(process)
             self.assertEqual(len(w), 1)
 
+    def test_no_logfile_and_no_syslog(self):
+        from supervisor.datatypes import boolean, logfile_name
```
Comment (mnaberez) [inline supervisor/tests/test_dispatchers.py:516]: No other existing tests in this file import things from `supervisor.datatypes` although they do similar configuration setup.

File: supervisor/tests/test_dispatchers.py [hunk 4]
```diff
@@ -512,6 +512,76 @@ def test_syslog_logfile_deprecated(self):
...
+        from supervisor.options import ProcessConfig, ServerOptions
+        options = ServerOptions() # need real options to get a real logger
+        options.loglevel = 20 # INFO, see loggers.py LevelsByName
```
Comment (mnaberez) [inline supervisor/tests/test_dispatchers.py:519]: Existing tests in this file do things like `options.loglevel = loggers.LevelsByName.TRAC`.  Why is this different?

**Comment by mnaberez** [comment]
Thanks for the excellent original report and this nice patch.  Some minor questions above.

File: supervisor/loggers.py (hunk 2 above)
Comment (scottkof) [reply supervisor/loggers.py:410]: I made this change because I needed a logger at dispatchers.py:154. Maybe loggers are only returned from the handle_ functions when they're needed?

File: supervisor/tests/test_dispatchers.py (hunk 3 above)
Comment (scottkof) [reply supervisor/tests/test_dispatchers.py:516]: Other tests don't read special LOGFILE_ datatypes (datatypes.py:91-92) into their configs. I used the datatypes functions so that the test config input (e.g. lines 521-22) is exactly what would be read from a real config file. I believe this makes the tests more readable and more robust.

File: supervisor/tests/test_dispatchers.py (hunk 4 above)
Comment (scottkof) [reply supervisor/tests/test_dispatchers.py:519]: This was an oversight. I'll change it.

File: supervisor/dispatchers.py (hunk 1 above)
Comment (scottkof) [reply supervisor/dispatchers.py:142]: The code now specially handles `logfile` being an empty string, in which case the behavior is the same as always.

**Comment by mnaberez** [comment]
Committed in ab1e07cc6d502c452f00c1e2d88989e91e75f294.  Thanks!
